apps/meteor/app/livechat/server/hooks/sendToCRM.ts

```diff
@@ -38,55 +76,48 @@ const getAdditionalFieldsByType = (type, room) => {
 			return {};
 	}
 };
-async function sendToCRM(type, room, includeMessages = true) {
+
+async function sendToCRM(
+	type: string,
+	room: OmnichannelRoomWithExtraFields,
+	includeMessages: boolean | IOmnichannelSystemMessage[] = true,
+): Promise<OmnichannelRoomWithExtraFields> {
 	if (!settings.get('Livechat_webhookUrl')) {
 		return room;
 	}
 
-	const postData = await Livechat.getLivechatRoomGuestInfo(room);
-
-	postData.type = type;
+	const postData: Awaited<ReturnType<typeof Livechat.getLivechatRoomGuestInfo>> & { type: string; messages: IOmnichannelSystemMessage[] } =
+		{
+			...(await Livechat.getLivechatRoomGuestInfo(room)),
+			type,
+			messages: [],
+		};
 
-	postData.messages = [];
-
-	let messages;
+	let messages: IOmnichannelSystemMessage[] | null = null;
 	if (typeof includeMessages === 'boolean' && includeMessages) {
-		messages = await Messages.findVisibleByRoomId(room._id, { sort: { ts: 1 } }).toArray();
+		messages = (await Messages.findVisibleByRoomId(room._id, { sort: { ts: 1 } }).toArray()) as IOmnichannelSystemMessage[];
```

Comment:
do you really need to cast is as `IOmnichannelSystemMessage` ? looks like you're properly testing it later, using like `isOmnichannelNavigationMessage`..

if really required, can you please improve the model implementation to support a generic so you could do like this:
```suggestion
		messages = await Messages.findVisibleByRoomId<IOmnichannelSystemMessage>(room._id, { sort: { ts: 1 } }).toArray();
```